Formula/tmux.rb

```diff
@@ -3,21 +3,20 @@ class Tmux < Formula
   homepage "https://tmux.github.io/"
 
   stable do
-    url "https://github.com/tmux/tmux/releases/download/2.1/tmux-2.1.tar.gz"
-    sha256 "31564e7bf4bcef2defb3cb34b9e596bd43a3937cad9e5438701a81a5a9af6176"
+    url "https://github.com/tmux/tmux/releases/download/2.2/tmux-2.2.tar.gz"
+    sha256 "bc28541b64f99929fe8e3ae7a02291263f3c97730781201824c0f05d7c8e19e4"
 
     patch do
-      # This fixes the Tmux 2.1 update that broke the ability to use select-pane [-LDUR]
-      # to switch panes when in a maximized pane https://github.com/tmux/tmux/issues/150#issuecomment-149466158
-      url "https://github.com/tmux/tmux/commit/a05c27a7e1c4d43709817d6746a510f16c960b4b.diff"
-      sha256 "2a60a63f0477f2e3056d9f76207d4ed905de8a9ce0645de6c29cf3f445bace12"
+      # required for the following unicode patch
+      url "https://github.com/tmux/tmux/commit/d303e5.patch"
+      sha256 "a3ae96b209254de9dc1f10207cc0da250f7d5ec771f2b5f5593c687e21028f67"
     end
 
     patch do
-      # This fixes the Tmux 2.1 update that breaks "tmux killw\; detach"
-      # https://github.com/tmux/tmux/issues/153#issuecomment-150184957
-      url "https://github.com/tmux/tmux/commit/3ebcf25149d75977ea97e9d4f786e0508d1a0d5e.diff"
-      sha256 "65a8bc0b2f6a8b41ad27605fd99419fff36314499969adc9d17dd3940a173508"
+      # workaround for bug in system unicode library reporting negative width
+      # for some valid characters
+      url "https://github.com/tmux/tmux/commit/23fdbc.patch"
```

Comment:
> I think the full commit is still preferable. Otherwise, brew fetch will fail if a new commit with same prefix is added by upstream. Also this is more consistent with style in rest of our code.

@xu-cheng `brew fetch` will never fail in that situation; GitHub will fail to deliver the patch. Six characters is more than enough to be unique in all but the largest Git histories. Eight characters is definitely sufficient. I don't think this is more consistent; I've always used shorter patch URLs when possible.